[Multiarch]Preflight fails with panic: runtime error: invalid memory address or nil pointer dereference for rpm-manifest for rhel9 images

## Bug Description
[Power9]Preflight fails with panic: runtime error: invalid memory address or nil pointer dereference for rpm-manifest
## Version and Command Invocation
preflight version 1.1.0-rc1 <commit: 0c0918de527faae53b5df9efb7a054af533d455c>

## Steps to Reproduce:
(How can we reproduce this?)
1) ./preflight-linux-ppc64le check container registry.access.redhat.com/rhel9-beta/support-tool:9.0.0beta-3
2)
3)

## Expected Result
Preflight should run the tests

## Actual Result
`[root@rdma-cert-03-lp7 ~]# ./preflight-linux-ppc64le check container registry.access.redhat.com/rhel9-beta/support-tools:9.0.0beta-3
time="2022-04-21T03:52:53-04:00" level=debug msg="config file not found, proceeding without it"
time="2022-04-21T03:52:53-04:00" level=info msg="certification library version 1.1.0-rc1 <commit: 0c0918de527faae53b5df9efb7a054af533d455c>"
time="2022-04-21T03:52:53-04:00" level=debug msg="target image: registry.access.redhat.com/rhel9-beta/support-tools:9.0.0beta-3"
time="2022-04-21T03:52:53-04:00" level=debug msg="pulling image from target registry"
time="2022-04-21T03:52:53-04:00" level=debug msg="temporary directory is /tmp/preflight-28931403"
time="2022-04-21T03:52:53-04:00" level=debug msg="exporting and flattening image"
time="2022-04-21T03:52:53-04:00" level=debug msg="extracting container filesystem to /tmp/preflight-28931403/fs"
time="2022-04-21T03:52:53-04:00" level=debug msg="writing container filesystem to output dir: /tmp/preflight-28931403/fs"
time="2022-04-21T03:53:06-04:00" level=trace msg="image config written to disk: /root/artifacts/cert-image.json"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x47e9fc]

goroutine 73 [running]:
database/sql.(*Rows).Next(0x0)
	/opt/hostedtoolcache/go/1.17.8/x64/src/database/sql/sql.go:2944 +0x2c
github.com/knqyf263/go-rpmdb/pkg/sqlite3.(*SQLite3).Read.func1(0xc00013e0c0, 0xc0004703b0)
	/home/runner/go/pkg/mod/github.com/opdev/go-rpmdb@v0.0.0-20220329221310-ca9c80a97804/pkg/sqlite3/sqlite3.go:67 +0x240
created by github.com/knqyf263/go-rpmdb/pkg/sqlite3.(*SQLite3).Read
	/home/runner/go/pkg/mod/github.com/opdev/go-rpmdb@v0.0.0-20220329221310-ca9c80a97804/pkg/sqlite3/sqlite3.go:52 +0x5c`

## Additional Context
(Anything else you think might help us troubleshoot, like your platform, dependency versions, etc).

--- a/.github/workflows/build-release.yml
+++ b/.github/workflows/build-release.yml
@@ -1,12 +1,10 @@
 name: Build Release
 
 on:
+  release:
+    types:
+      - published
   push:
-    tags:
-      - '[0-9]+.[0-9]+.[0-9]+'
-      - '[0-9]+.[0-9]+.[0-9]+-alpha[0-9]+'
-      - '[0-9]+.[0-9]+.[0-9]+-beta[0-9]+'
-      - '[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
     branches: [ ghactions ]
 
 env:
@@ -68,3 +66,14 @@ jobs:
       registry: ${{ secrets.IMAGE_REGISTRY }}
       user: ${{ secrets.REGISTRY_USER }}
       password: ${{ secrets.REGISTRY_PASSWORD }}
+
+  extract-assets:
+    needs: build-release
+    uses: ./.github/workflows/release-artifacts.yml
+
+    with:
+      name: ${{ needs.build-release.outputs.imageName }}
+      tag: ${{ needs.build-release.outputs.imageVersion }}
+    secrets:
+      registry: ${{ secrets.IMAGE_REGISTRY }}
+      token: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/release-artifacts.yml
+++ b/.github/workflows/release-artifacts.yml
@@ -1,35 +1,50 @@
 name: Create Release Artifacts
 
 on:
-  release:
-    types:
-      - created
-
+  workflow_call:
+    inputs:
+      name:
+        required: true
+        type: string
+      tag:
+        required: true
+        type: string
+    secrets:
+      registry:
+        required: true
+      token:
+        required: true
+        
 jobs:
-  build-binaries:
+  release-assets-matrix:
+    name: Release Matrix
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [ linux ]
+        architecture:
+          - amd64
+          - arm64
+          - ppc64le
+          - s390x
     steps:
-
-    - name: code checkout
-      uses: actions/checkout@v2
-
-    - name: Set Env Tags
-      run: echo RELEASE_TAG=$(echo $GITHUB_REF | cut -d '/' -f 3) >> $GITHUB_ENV
-
-    - name: Install system deps
-      run: 'sudo apt update && sudo apt install -y libgpgme-dev libbtrfs-dev libdevmapper-dev'
-
-    - name: Set up Go
-      uses: actions/setup-go@v2
+    - name: Extract executable
+      uses: shrink/actions-docker-extract@v1
+      id: extract
       with:
-        go-version: 1.17
-    
-    - name: Build Multi-arch binaries
-      id: build-multi-arch-binaries
-      run: make build-multi-arch
+        image: ${{ secrets.registry }}/${{ inputs.name }}:${{ inputs.tag }}-${{ matrix.platform }}-${{ matrix.architecture }}
+        path: /usr/local/bin/preflight
+
+    - name: Rename the binary
+      uses: canastro/copy-file-action@master
+      with:
+        source: ${{ steps.extract.outputs.destination }}/preflight
+        target: ${{ steps.extract.outputs.destination }}/preflight-${{ matrix.platform }}-${{ matrix.architecture }}
 
     - name: Upload binaries to the release
       uses: AButler/upload-release-assets@v2.0
+      id: upload-release-asset
       with:
-        files: "preflight*"
-        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        files: ${{ steps.extract.outputs.destination }}/preflight-${{ matrix.platform }}-${{ matrix.architecture }}
+        repo-token: ${{ secrets.token }}
+        release-tag: ${{ inputs.tag }}

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 .DEFAULT_GOAL:=help
 
-BINARY=preflight
+BINARY?=preflight
 IMAGE_BUILDER?=podman
 IMAGE_REPO?=quay.io/opdev
 VERSION=$(shell git rev-parse HEAD)

